pages/api/delete-poap-event.js

```diff
@@ -0,0 +1,22 @@
+import withMethods from '../../util/api/withMethods'
+import withAuth from '../../util/api/withAuth'
+import withValidParams from '../../util/api/withValidParams'
+import { deleteCommunityPoapEvent } from '../../util/api/fauna'
+
+const handler = async (req, res) => {
```

Comment:
There is a problem with this endpoint in that any authenticated user can delete any event. It looks like the previous endpoints would sort of do an authorization check by inferring the community from the authenticated address with each handler starting with this:
```
const { _id } = await `(address)
```

I could work around this by fetching the community associated with the event plus adding the `findCommunityByAddress` and the comparing in the handler but I'm not sure this is the authorization pattern we want to establish. It requires a lot more code in each API handler and more round trips to Fauna.

As we add more features and data models, I think we will be well served by changing our authorization approach a bit so that we can take advantage of Fauna's [Attribute-based Access Control](https://docs.fauna.com/fauna/current/security/abac). I could open a separate PR that demonstrates how we might do that. Thoughts?